### PR TITLE
Public release of the UVCGAN2 code

### PR DESCRIPTION
**Pull request by usert5432**
Hi everyone,

I am in the process of finalizing the UVCGAN2 code release. The codebase is complete, there are just a few bookkeeping tasks left. So, I am inviting everyone to start the review (@brettviren, @YHRen, @pphuangyi). 

To test the package you can use a conda env file `contrib/conda_env.yaml`. A few highlights of this PR:
- as a starting point I have taken tag `v0.0.3`  of toygan
- as a CycleGAN implementation I have taken `Batch-Head-CycleGAN`, removed parts that were not used in the uvcgan2 paper (HC, GP cacher, inpainting), and named it `UVCGAN2`.
- `v0.0.3` has tuning and ablation scripts. I have removed them and instead providing standalone cmdline scripts to train networks separately
- pre-trained models are available on `wavelet` under `/data/dtorbunov/results/2023-04-04_uvcgan2_pretrained_models`. I will upload them once the PR is ready
- as commit authors, I have indicated myself, Yi, Ray and Brett. Please, let me know if you would like me to add someone else.

Remaining TODO items:
- [x] add LICENSE headers to the original CycleGAN files
~~-  maybe add a nice image grid for the README(?)~~
- [x] add a data download script
- [x] write README
- [x] upload pre-trained models.
- [x] backport https://github.com/LS4GAN/toygan/pull/84 once it is done
- [ ] test everything (I will need **your** help with testing)

### CODE REVIEW

File: uvcgan2/cgan/uvcgan2.py
Comment (YHRen) [inline uvcgan2/cgan/uvcgan2.py]: can `self.gradient_penalty` be removed? There is also a `self.gp` which is an actual module. Maybe change `self.gradient_penalty` to `self.gp_cfg` (or `gp_config`) to better reflect its type or purpose.

**Review by YHRen — APPROVED** [review]
@usert5432 I've reviewed `uvcgan2.py`. everything looks good to me. Thank you!

**Comment by usert5432** [comment]
Pre-trained models have been uploaded to: https://zenodo.org/record/7813795

File: uvcgan2/cgan/uvcgan2.py
Comment (usert5432) [reply uvcgan2/cgan/uvcgan2.py]: Thank you, @YHRen! Fixed in fc36adfb39360437c63da0fbfb37d9dbe191e004.

**Review by YHRen — APPROVED** [review]
@usert5432 Looks good. Thank you!

**Comment by usert5432** [comment]
> @usert5432 Looks good. Thank you!

Thank you, @YHRen! Would you prefer for us to wait until the testing is finished, or merge PR now, and fix issues (if any) later?

**Comment by YHRen** [comment]
> > @usert5432 Looks good. Thank you!
> 
> Thank you, @YHRen! Would you prefer for us to wait until the testing is finished, or merge PR now, and fix issues (if any) later?

@usert5432 
I'm thinking to release it now and put a banner saying something like "under testing, and issue tickets are welcome". What do you think?

**Pull request merged**
